pkcs11/src/new/types/mechanism/mod.rs

```diff
@@ -339,356 +363,397 @@ pub enum Mechanism {
     RsaPkcsTpm_1_1,
     RsaPkcsOaepTpm_1_1,
 }
+*/
+
+impl From<MechanismType> for CK_MECHANISM_TYPE {
+    fn from(mechanism_type: MechanismType) -> Self {
+        match mechanism_type {
+            RSA_PKCS_KEY_PAIR_GEN => CKM_RSA_PKCS_KEY_PAIR_GEN,
```

Comment:
Why do you need to do this since you have `Deref` above and the value on the right is already inside the value on the left?

Reply:
ah you mean I could just do `&mechanism_type`? That's true

Reply:
Something like that, especially if you're guaranteed to get a good result